numba/cuda/tests/cudapy/test_const_string.py

```diff
@@ -52,5 +56,44 @@ def test_const_string(self):
         self.assertEqual(len(matches), 1)
 
 
+class TestConstString(CUDATestCase):
+    def test_assign_const_string(self):
+        # There is no NVVM on cudasim, but we might as well run this test.
+        if not config.ENABLE_CUDASIM and not NVVM().is_nvvm70:
+            self.skipTest('Character sequences unsupported on NVVM 3.4')
+
+        # Inspired by the reproducer from Issue #7041.
+
+        @cuda.jit
+        def str_assign(arr):
+            i = cuda.grid(1)
+            if i < len(arr):
+                arr[i] = "XYZ"
+
+        n_strings = 8
+        arr = np.zeros(n_strings + 1, dtype="<U12")
+        str_assign[1, n_strings](arr)
+
+        # Expected result, e.g.:
+        #     ['XYZ' 'XYZ' 'XYZ' 'XYZ' 'XYZ' 'XYZ' 'XYZ' 'XYZ' '']
+        expected = np.zeros_like(arr)
+        expected[:-1] = 'XYZ'
+        expected[-1] = ''
+        np.testing.assert_equal(arr, expected)
+
+    @skip_on_cudasim('Const strings not disallowed on cudasim')
+    def test_const_string_disallowed_nvvm34(self):
+        if NVVM().is_nvvm70:
+            self.skipTest('Character sequences are permitted with NVVM 7.0')
+
+        @cuda.jit
+        def str_assign(arr):
+            arr[0] = "XYZ"
+
+        arr = np.zeros(1, dtype="<U12")
+        with self.assertRaisesRegex(TypingError, '.*is a char sequence.*'):
+            str_assign[1, 1](arr)
```

Comment:
Suggest also testing with a non-unicode variant like `dtype="S4"` and `arr[0] = b"XYZ"`